### PR TITLE
Update with framework bindable changes

### PR DESCRIPTION
**Pull request by smoogipoo**
1. All methods marked as `updateX` use the pattern `.ValueChanged += e => updateX(e.NewValue)`
2. All methods marked as `XChanged` use the pattern `void XChanged(ValueChangedEvent<T> e)`

### CODE REVIEW

**Comment by smoogipoo** [comment]
Note: Build will compile with warnings until https://github.com/ppy/osu/pull/4300 is also merged.

File: osu.Game.Rulesets.Mania/Objects/Drawables/Pieces/NotePiece.cs [hunk 1]
```diff
@@ -49,9 +49,9 @@ public NotePiece()
         private void load(IScrollingInfo scrollingInfo)
         {
             direction.BindTo(scrollingInfo.Direction);
-            direction.BindValueChanged(direction =>
+            direction.BindValueChanged(e =>
             {
-                colouredBox.Anchor = colouredBox.Origin = direction == ScrollingDirection.Up ? Anchor.TopCentre : Anchor.BottomCentre;
+                colouredBox.Anchor = colouredBox.Origin = direction.Value == ScrollingDirection.Up ? Anchor.TopCentre : Anchor.BottomCentre;
```
Comment (peppy) [inline osu.Game.Rulesets.Mania/Objects/Drawables/Pieces/NotePiece.cs]:
```suggestion
                colouredBox.Anchor = colouredBox.Origin = e.NewValue == ScrollingDirection.Up ? Anchor.TopCentre : Anchor.BottomCentre;
```

File: osu.Game.Rulesets.Mania/UI/Components/ColumnBackground.cs [hunk 2]
```diff
@@ -48,9 +48,9 @@ private void load(IBindable<ManiaAction> action, IScrollingInfo scrollingInfo)
             };
 
             direction.BindTo(scrollingInfo.Direction);
-            direction.BindValueChanged(direction =>
+            direction.BindValueChanged(e =>
             {
-                backgroundOverlay.Anchor = backgroundOverlay.Origin = direction == ScrollingDirection.Up ? Anchor.TopLeft : Anchor.BottomLeft;
+                backgroundOverlay.Anchor = backgroundOverlay.Origin = direction.Value == ScrollingDirection.Up ? Anchor.TopLeft : Anchor.BottomLeft;
```
Comment (peppy) [inline osu.Game.Rulesets.Mania/UI/Components/ColumnBackground.cs]:
```suggestion
                backgroundOverlay.Anchor = backgroundOverlay.Origin = e.NewValue == ScrollingDirection.Up ? Anchor.TopLeft : Anchor.BottomLeft;
```

File: osu.Game.Rulesets.Mania/UI/Components/ColumnHitObjectArea.cs [hunk 3]
```diff
@@ -49,9 +49,9 @@ public ColumnHitObjectArea(HitObjectContainer hitObjectContainer)
         private void load(IScrollingInfo scrollingInfo)
         {
             direction.BindTo(scrollingInfo.Direction);
-            direction.BindValueChanged(direction =>
+            direction.BindValueChanged(e =>
             {
-                Anchor anchor = direction == ScrollingDirection.Up ? Anchor.TopLeft : Anchor.BottomLeft;
+                Anchor anchor = direction.Value == ScrollingDirection.Up ? Anchor.TopLeft : Anchor.BottomLeft;
```
Comment (peppy) [inline osu.Game.Rulesets.Mania/UI/Components/ColumnHitObjectArea.cs]:
```suggestion
                Anchor anchor = e.NewValue == ScrollingDirection.Up ? Anchor.TopLeft : Anchor.BottomLeft;
```

File: osu.Game.Rulesets.Mania/UI/Components/ColumnKeyArea.cs [hunk 4]
```diff
@@ -64,11 +64,11 @@ private void load(IBindable<ManiaAction> action, IScrollingInfo scrollingInfo)
             };
 
             direction.BindTo(scrollingInfo.Direction);
-            direction.BindValueChanged(direction =>
+            direction.BindValueChanged(e =>
             {
                 gradient.Colour = ColourInfo.GradientVertical(
-                    direction == ScrollingDirection.Up ? Color4.Black : Color4.Black.Opacity(0),
-                    direction == ScrollingDirection.Up ? Color4.Black.Opacity(0) : Color4.Black);
+                    direction.Value == ScrollingDirection.Up ? Color4.Black : Color4.Black.Opacity(0),
```
Comment (peppy) [inline osu.Game.Rulesets.Mania/UI/Components/ColumnKeyArea.cs]:
```suggestion
                    e.NewValue == ScrollingDirection.Up ? Color4.Black : Color4.Black.Opacity(0),
```

File: osu.Game.Rulesets.Mania/UI/Components/ColumnKeyArea.cs [hunk 5]
```diff
@@ -64,11 +64,11 @@ private void load(IBindable<ManiaAction> action, IScrollingInfo scrollingInfo)
...
+                    direction.Value == ScrollingDirection.Up ? Color4.Black.Opacity(0) : Color4.Black);
```
Comment (peppy) [inline osu.Game.Rulesets.Mania/UI/Components/ColumnKeyArea.cs]:
```suggestion
                    e.NewValue == ScrollingDirection.Up ? Color4.Black.Opacity(0) : Color4.Black);
```

File: osu.Game.Rulesets.Osu/UI/Cursor/GameplayCursor.cs [hunk 6]
```diff
@@ -183,13 +183,13 @@ private void load(OsuConfigManager config, IBindable<WorkingBeatmap> beatmap)
                 };
 
                 this.beatmap.BindTo(beatmap);
-                this.beatmap.ValueChanged += v => calculateScale();
+                this.beatmap.ValueChanged += e => calculateScale();
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/UI/Cursor/GameplayCursor.cs]:
```suggestion
                this.beatmap.ValueChanged += _ => calculateScale();
```

File: osu.Game.Rulesets.Osu/UI/Cursor/GameplayCursor.cs [hunk 7]
```diff
@@ -183,13 +183,13 @@ private void load(OsuConfigManager config, IBindable<WorkingBeatmap> beatmap)
...
 
                 cursorScale = config.GetBindable<double>(OsuSetting.GameplayCursorSize);
-                cursorScale.ValueChanged += v => calculateScale();
+                cursorScale.ValueChanged += e => calculateScale();
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/UI/Cursor/GameplayCursor.cs]:
```suggestion
                cursorScale.ValueChanged += _ => calculateScale();
```

File: osu.Game.Rulesets.Osu/UI/Cursor/GameplayCursor.cs [hunk 8]
```diff
@@ -183,13 +183,13 @@ private void load(OsuConfigManager config, IBindable<WorkingBeatmap> beatmap)
...
 
                 autoCursorScale = config.GetBindable<bool>(OsuSetting.AutoCursorSize);
-                autoCursorScale.ValueChanged += v => calculateScale();
+                autoCursorScale.ValueChanged += e => calculateScale();
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/UI/Cursor/GameplayCursor.cs]:
```suggestion
                autoCursorScale.ValueChanged += _ => calculateScale();
```

File: osu.Game.Tests/Visual/TestCaseDrawableDate.cs [hunk 9]
```diff
@@ -60,7 +60,7 @@ public PokeyDrawableDate(DateTimeOffset date)
                     }
                 };
 
-                drawableDate.Current.ValueChanged += v => flash.FadeOutFromOne(500);
+                drawableDate.Current.ValueChanged += e => flash.FadeOutFromOne(500);
```
Comment (peppy) [inline osu.Game.Tests/Visual/TestCaseDrawableDate.cs]:
```suggestion
                drawableDate.Current.ValueChanged += _ => flash.FadeOutFromOne(500);
```

File: osu.Game/Beatmaps/WorkingBeatmap.cs [hunk 10]
```diff
@@ -36,7 +36,7 @@ protected WorkingBeatmap(BeatmapInfo beatmapInfo)
             BeatmapSetInfo = beatmapInfo.BeatmapSet;
             Metadata = beatmapInfo.Metadata ?? BeatmapSetInfo?.Metadata ?? new BeatmapMetadata();
 
-            Mods.ValueChanged += mods => applyRateAdjustments();
+            Mods.ValueChanged += e => applyRateAdjustments();
```
Comment (peppy) [inline osu.Game/Beatmaps/WorkingBeatmap.cs]:
```suggestion
            Mods.ValueChanged += _ => applyRateAdjustments();
```

File: osu.Game/Overlays/BeatmapSet/Header.cs [hunk 11]
```diff
@@ -223,7 +223,7 @@ private void updateDownloadButtons()
             {
                 case DownloadState.LocallyAvailable:
                     // temporary for UX until new design is implemented.
-                    downloadButtonsContainer.Child = new osu.Game.Overlays.Direct.DownloadButton(BeatmapSet)
+                    downloadButtonsContainer.Child = new osu.Game.Overlays.Direct.DownloadButton(BeatmapSet.Value)
```
Comment (peppy) [inline osu.Game/Overlays/BeatmapSet/Header.cs]:
```suggestion
                    downloadButtonsContainer.Child = new osu.Game.Overlays.Direct.DownloadButton(e.NewValue)
```

File: osu.Game/Overlays/BeatmapSet/Header.cs [hunk 12]
```diff
@@ -232,12 +232,12 @@ private void updateDownloadButtons()
                 case DownloadState.Downloading:
                 case DownloadState.Downloaded:
                     // temporary to avoid showing two buttons for maps with novideo. will be fixed in new beatmap overlay design.
-                    downloadButtonsContainer.Child = new DownloadButton(BeatmapSet);
+                    downloadButtonsContainer.Child = new DownloadButton(BeatmapSet.Value);
```
Comment (peppy) [inline osu.Game/Overlays/BeatmapSet/Header.cs]:
```suggestion
                    downloadButtonsContainer.Child = new DownloadButton(e.NewValue);
```

File: osu.Game/Overlays/BeatmapSet/Header.cs [hunk 13]
```diff
@@ -232,12 +232,12 @@ private void updateDownloadButtons()
...
                     break;
                 default:
-                    downloadButtonsContainer.Child = new DownloadButton(BeatmapSet);
+                    downloadButtonsContainer.Child = new DownloadButton(BeatmapSet.Value);
```
Comment (peppy) [inline osu.Game/Overlays/BeatmapSet/Header.cs]:
```suggestion
                    downloadButtonsContainer.Child = new DownloadButton(e.NewValue);
```

File: osu.Game/Overlays/BeatmapSet/Header.cs [hunk 14]
```diff
@@ -232,12 +232,12 @@ private void updateDownloadButtons()
...
                     if (BeatmapSet.Value.OnlineInfo.HasVideo)
-                        downloadButtonsContainer.Add(new DownloadButton(BeatmapSet, true));
+                        downloadButtonsContainer.Add(new DownloadButton(BeatmapSet.Value, true));
```
Comment (peppy) [inline osu.Game/Overlays/BeatmapSet/Header.cs]:
```suggestion
                        downloadButtonsContainer.Add(new DownloadButton(e.NewValue, true));
```

File: osu.Game/Overlays/Music/PlaylistItem.cs [hunk 15]
```diff
@@ -100,7 +100,7 @@ private void load(OsuColour colours, LocalisationManager localisation)
             titleBind = localisation.GetLocalisedString(new LocalisedString((metadata.TitleUnicode, metadata.Title)));
             artistBind = localisation.GetLocalisedString(new LocalisedString((metadata.ArtistUnicode, metadata.Artist)));
 
-            artistBind.BindValueChanged(newText => recreateText(), true);
+            artistBind.BindValueChanged(e => recreateText(), true);
```
Comment (peppy) [inline osu.Game/Overlays/Music/PlaylistItem.cs]:
```suggestion
            artistBind.BindValueChanged(_ => recreateText(), true);
```

File: osu.Game/Overlays/Settings/SettingsItem.cs [hunk 16]
```diff
@@ -119,7 +119,7 @@ public Bindable<T> Bindable
                 set
                 {
                     bindable = value;
-                    bindable.ValueChanged += newValue => UpdateState();
+                    bindable.ValueChanged += e => UpdateState();
```
Comment (peppy) [inline osu.Game/Overlays/Settings/SettingsItem.cs]:
```suggestion
                    bindable.ValueChanged += _ => UpdateState();
```

File: osu.Game/Overlays/Settings/SettingsItem.cs [hunk 17]
```diff
@@ -119,7 +119,7 @@ public Bindable<T> Bindable
...
                     bindable.DisabledChanged += disabled => UpdateState();
```
Comment (peppy) [inline osu.Game/Overlays/Settings/SettingsItem.cs]:
```suggestion
                    bindable.DisabledChanged += _ => UpdateState();
```

File: osu.Game/Overlays/SocialOverlay.cs [hunk 18]
```diff
@@ -67,18 +67,18 @@ public SocialOverlay()
                 }
             };
 
-            Header.Tabs.Current.ValueChanged += tab => Scheduler.AddOnce(updateSearch);
+            Header.Tabs.Current.ValueChanged += e => Scheduler.AddOnce(updateSearch);
```
Comment (peppy) [inline osu.Game/Overlays/SocialOverlay.cs]:
```suggestion
            Header.Tabs.Current.ValueChanged += _ => Scheduler.AddOnce(updateSearch);
```

File: osu.Game/Overlays/SocialOverlay.cs [hunk 19]
```diff
@@ -67,18 +67,18 @@ public SocialOverlay()
...
 
-            Filter.Tabs.Current.ValueChanged += sortCriteria => Scheduler.AddOnce(updateSearch);
+            Filter.Tabs.Current.ValueChanged += e => Scheduler.AddOnce(updateSearch);
```
Comment (peppy) [inline osu.Game/Overlays/SocialOverlay.cs]:
```suggestion
            Filter.Tabs.Current.ValueChanged += _ => Scheduler.AddOnce(updateSearch);
```

File: osu.Game/Screens/Multi/Match/Components/MatchChatDisplay.cs [hunk 20]
```diff
@@ -28,13 +28,13 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
-            roomId.BindValueChanged(v => updateChannel(), true);
+            roomId.BindValueChanged(e => updateChannel(), true);
```
Comment (peppy) [inline osu.Game/Screens/Multi/Match/Components/MatchChatDisplay.cs]:
```suggestion
            roomId.BindValueChanged(_ => updateChannel(), true);
```

**Review by peppy — CHANGES_REQUESTED** [review]

**Comment by peppy** [comment]
Proposal #1:

Use the previous variable names instead of `e`. I think this will read better (`channel.NewValue` rather than `e.NewValue`).

Proposal #2:

Give ValueChangedEvent implicit operator that returns `NewValue` by default, so we don't need the NewValue suffix in most cases. Not sure about this one, but seems like it may tidy this back up?

**Review by peppy — DISMISSED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**